### ISSUE TITLE
safety.dat unable to SEEK_SET to 0

### ISSUE THREAD

**Issue by heifner** [AntelopeIO/spring#646]
Noticed in log of `lib_advance_test.py`

This was before #624 
https://github.com/AntelopeIO/spring/actions/runs/10564544450

```
warn  2024-08-26T18:00:00.910 net-1     finalizer.cpp:209             save_finalizer_safet ] 13 NSt8ios_base7failureB5cxx11E: cfile: /__w/spring/spring/build/TestLogs/lib_advance_test9/node_03/finalizers/safety.dat unable to SEEK_SET to: 0, ferror: 0: iostream error
rethrow cfile: /__w/spring/spring/build/TestLogs/lib_advance_test9/node_03/finalizers/safety.dat unable to SEEK_SET to: 0, ferror: 0: iostream error: 
    {"what":"cfile: /__w/spring/spring/build/TestLogs/lib_advance_test9/node_03/finalizers/safety.dat unable to SEEK_SET to: 0, ferror: 0: iostream error"}
    net-1  finalizer.cpp:209 save_finalizer_safety_info
```

**Comment by bhazzard** [comment]
catch and stop voting, possibly shut down.

**Comment by heifner** [comment]
Was able to reproduce this issue on branch f866521044, but not on branch 5315ce679a5f3519552582a688c02c1f62d434c8. Appears to have been fixed by #624 somehow. Unclear as nothing in #624 should have made any difference in this regard.
